### PR TITLE
[WIP] SPLADE & BM25 Notebooks

### PR DESCRIPTION
**Pull request by gdj0nes**
Introduce two new notebooks for working with splade and bm25

### CODE REVIEW

**Review by jamescalam — CHANGES_REQUESTED** [review]

File: sparse/bm25/bm25-vector-generation.ipynb [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sentence_transformers import SentenceTransformer\n",
+    "\n",
+    "model = SentenceTransformer('sentence-transformers/all-MiniLM-L6-v2')"
```
[the remark this answers is not in the document]
Comment (gdj0nes) [reply sparse/bm25/bm25-vector-generation.ipynb]: Added to the notebooks

File: sparse/bm25/bm25-vector-generation.ipynb [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture\n",
+    "!python -m spacy download en_core_web_sm"
```
[the remark this answers is not in the document]
Comment (gdj0nes) [reply sparse/bm25/bm25-vector-generation.ipynb:128]: this didn't have an issue when i ran it in colab just now

File: sparse/bm25/bm25-vector-generation.ipynb [hunk 3]
```diff
@@ -0,0 +1,430 @@
+{
```
Comment (acatav) [inline sparse/bm25/bm25-vector-generation.ipynb:1]: <p>We need to load the original public available qoura dataset. This dataset is coming with pairs of qoura questions and a label, we need to add here code that flatten it to single question per row.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/132/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (acatav) [inline sparse/bm25/bm25-vector-generation.ipynb:1]: <p>We need to use batches here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/132/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (acatav) [inline sparse/bm25/bm25-vector-generation.ipynb:1]: <p>This notebook should not upsert to Pinecone at all, only show how to generate the data</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/132/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

File: sparse/bm25/bm25-vector-generation.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (acatav) [reply sparse/bm25/bm25-vector-generation.ipynb:128]: It happens to me all the time, so I guess we should try NLTK and see if it installed more smoothly

[the remark this answers is not in the document]
Comment (gdj0nes) [reply sparse/bm25/bm25-vector-generation.ipynb:128]: We can't unless we recompute all of the datasets

File: sparse/bm25/bm25-vector-generation.ipynb (hunk 3 above)
Comment (gdj0nes) [reply sparse/bm25/bm25-vector-generation.ipynb:1]: What? People need to know how to use this product in a concise example

**Pull request closed without merging**